### DISCUSSION TITLE
Git push became slow after adding git-lfs

### DISCUSSION THREAD

**Discussion by siddhsql** [git-lfs/git-lfs#5793] [General]
I added git-lfs to a repository. I am tracking a few large files about 10-50MB each. After adding git-lfs my `git push` to remote repository (GitHub) runs slow even though no changes are made to files tracked by git-lfs. here are some details:

```
╰─➤  git --version
git version 2.39.2 (Apple Git-143)
```

```
╰─➤  git-lfs --version
git-lfs/3.5.1 (GitHub; darwin arm64; go 1.21.7; git e237bb3a)
```

```
╰─➤  git lfs fsck
Git LFS fsck OK
```

```
╰─➤  git lfs status
On branch mychanges

Objects to be committed:


Objects not staged for commit:

```

^^ this is key - even with clean status my `git push` runs slow. Is there any way I can fix this? what is going on here?

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're experiencing a problem.  You could try running the `git push` with `GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1` to possibly get a better sense from the timing data in the trace of where the delay occurs.

For instance, if you are pushing a large number of changed files, even if they are not files tracked by Git LFS, the `pre-push` Git LFS hook has to search through them in case any are Git LFS pointers and the corresponding Git LFS objects also need to be pushed.  (This can be skipped with `GIT_LFS_SKIP_PUSH=1`, but you'll then need to use `git lfs push --all` to push up the Git LFS objects.)

**Comment by siddhsql** [comment]
Hi Chris,

I ran that command and it did show me a bunch of git-lfs stuff. But in the
process I actually made a mistake. I had a rule to track all files with a
certain extension and it ended up tracking a file it shouldn't and pushed
it to GitHub. I want to undo this operation now. I'll start a new thread
for this as this is a more urgent problem for me now.
